azkaban-common/src/main/java/azkaban/executor/ExecutorHealthChecker.java

```diff
@@ -87,11 +90,27 @@ public void shutdown() {
     }
   }
 
+  /**
+   * Wrapper for capturing and logging errors and exceptions thrown during healthcheck.
+   * {@code ScheduledExecutorService} stops the scheduled invocations of a given method in
+   * case it throws an exception.
+   * Any exceptions are not expected at this stage however in case any errors and unchecked
+   * exceptions do occur, we still don't want subsequent healthchecks to stop.
+   */
+  public void checkExecutorHealthQuietly() {
+    try {
+      checkExecutorHealth();
+    } catch (Throwable t) {
```

Comment:
I feel we need to differentiate between Runtime Exception and java.lang.Error out here. We need to only handle Runtime Exceptions. For java.lang.Error shouldn't we stop the server or raise an alert? (Handling if java.lang.Error can be done as a follow-on, and tracked by filing a ticket).

Reply:
Will do.